docs/tools/offline/cli.md

```diff
@@ -71,241 +62,206 @@ $ python setup.py install
-
-##### Uninstalling Mbed CLI
-
-To uninstall Mbed CLI, run:
+### Configuring Mbed CLI
 
-```
-pip uninstall mbed-cli
-```
+There are some configuration that you must set before you can work with Mbed CLI.
 
-##### Adding Bash tab completion
+#### Mandatory: Setting PATH variables
 
-To install `mbed-cli` bash tab completion navigate to the `tools/bash_completion` directory. Then, copy the `mbed` script into your `/etc/bash_completion.d/` or `/usr/local/etc/bash_completion.d` directory and reload your terminal.  
-
-[Full documentation here](https://github.com/ARMmbed/mbed-cli/blob/master/tools/bash_completion/install.md)
+Mbed CLI requires adding the following to the system `PATH`:
 
-#### Quickstart video
+* The paths for the Git and Mercurial executables (`git` and `hg`).
 
-<span class="images">[![Video tutorial](https://img.youtube.com/vi/PI1Kq9RSN_Y/0.jpg)](https://www.youtube.com/watch?v=PI1Kq9RSN_Y)</span>
+#### Mandatory: Toolchain selection
 
-#### Before you begin: understanding the working context and program root
+You need to tell Mbed CLI where to find the toolchains that you want to use for compiling your source tree.
 
-Mbed CLI uses the current directory as a working context, in a similar way to Git, Mercurial and many other command-line tools. This means that before calling any Mbed CLI command, you must first change to the directory containing the code you want to act on. For example, if you want to update the Mbed OS sources in your `mbed-example-program` directory:
+There are multiple ways to configure toolchain locations:
 
-```
-$ cd mbed-example-program
-$ cd mbed-os
-$ mbed update master   # This will update "mbed-os", not "my-program"
-```
+* The `mbed_settings.py` file in the root of your program. The tools will automatically create this file if it doesn't already exist.
+* The Mbed CLI configuration command.
+* Setting an environment variable.
+* Adding the compiler's directory to your PATH.
 
-Various Mbed CLI features require a program root, which should be under version control - either [Git](https://git-scm.com/) or [Mercurial](https://www.mercurial-scm.org/). This makes it possible to switch between revisions of the whole program and its libraries, control the program history, synchronize the program with remote repositories, share it with others and so on. Version control is also the primary and preferred delivery mechanism for Mbed OS source code, which allows everyone to contribute to Mbed OS.
+Methods for configuring toolchains that appear earlier in the above list override methods that appear later.
 
-<span class="warnings">**Warning**: Mbed CLI stores information about libraries and dependencies in reference files that use the `.lib` extension (such as `lib_name.lib`). Although these files are human-readable, we *strongly* advise that you don't edit these manually - let Mbed CLI manage them instead.</span>
+##### Through `mbed_settings.py`
```

Comment:
Perhaps move this as a 2nd section. We'd like to encourage developers to use the Mbed CLI configuration capability. Also it's highly likely that this is deprecated in the future from the build tools (cc @theotherjimmy)

Reply:
@AnotherButler @screamerbg Yes this is not a recommended method for configuring a toolchain. It will work at the moment, so I would like to keep it in. Maybe mentioning the fact that we prefer other methods is in order?

Reply:
I agree that we should organise this by how we want people to use it

Reply:
We have a bit of a problem, as each member in the list has a priority, in descending order, and some of them are deprecated/discouraged. I welcome any recommendations on how we can still convey the priority, while writing our preferred method first.

For reference, my preference is:
 * Mbed CLI Config
 * PATH
 * Environment Variables
 * `mbed_settings.py`

And the priority (highest priority first) is:
 * `mbed_settings.py`
 * Mbed CLI Config
 * Environment Variables
 * PATH